src/platforms/hooks/usePlatformUsageInfo.ts

```diff
@@ -0,0 +1,42 @@
+import useGuild from "components/[guild]/hooks/useGuild"
+import useSWRWithOptionalAuth from "hooks/useSWRWithOptionalAuth"
+import platforms, { PlatformUsageRestrictions } from "platforms/platforms"
+import { Guild, PlatformName } from "types"
+
+const usePlatformUsageInfo = (
+  platform: PlatformName,
+  platformId: string
+): { isAlreadyInUse: boolean; guildUrlName?: string; isValidating: boolean } => {
+  const { usageRestriction } = platforms[platform]
+
+  const { id } = useGuild()
+
+  // Not sure if it'll be a problem (API performance wise) that we call this endpoint here, but
```

Comment:
Maybe Balazs can provide some info regarding this, then we can remove this comment ;)

Reply:
I think the fact that we request them one by one shouldn't be a big problem, I think we did it previously as well. Two think I'm not sure about:
- Do we need it to be mutable?
- We should avoid resending requests that resulted in 404. I could think of these two possible solutions:
  1) Map 404 errors to successes in this SWR call's fetcher. A bit messy, since we need to pass the fetcher manually
  2) Disable `shouldRetryOnError`
  - In the latter case, we might run into cases where a revalidation might be necessary, therefore the other solution might be better. That way we only "ignore" the 404 responses, which is fine, since from a 404 response, we already know all the information we needed. Let me know which approach you prefer, or if you have an alternative!

Reply:
Thanks for the ideas! I made it immutable & decided to just disable `shouldRetryOnError` for now, but we can talk more about it on DC!